yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -188,6 +188,27 @@ describe('AVM simulator: transpiled Noir contracts', () => {
 
     expect(results.reverted).toBe(false);
   });
+
+  it('execution of a non-existent contract immediately reverts', async () => {
+    const context = initContext();
+    const startGas = { l2Gas: context.machineState.l2GasLeft, daGas: context.machineState.daGasLeft };
+    const results = await new AvmSimulator(context).execute();
+
+    expect(results.reverted).toBe(true);
+    expect(results.output).toEqual([]);
+    expect(results.gasLeft).toEqual(startGas);
```

Comment:
I'm curious _why_ Ethereum uses the entire gas allocation when a call reverts instead of just charging whatever was actually used. This is a case where we copied Ethereum, but I don't actually know why EVM does that.